src/containers/BookList.js

```diff
@@ -18,12 +39,17 @@ BookList.propTypes = {
       category: PropTypes.string.isRequired,
     }),
   ).isRequired,
+  delete: PropTypes.func.isRequired,
+  handleVisibility: PropTypes.func.isRequired,
+  handleChangeFilter: PropTypes.func.isRequired,
 };
 
 const mapStateToProps = state => ({ books: state.books });
 
 const mapDispatchToProps = dispatch => ({
   delete: book => dispatch(deleteBook(book)),
+  handleChangeFilter: filter => (dispatch({ type: CHANGE_FILTER, filter })),
+  handleVisibility: filter => (dispatch({ type: CHANGE_VISIBILITY, filter })),
 });
 
 export default connect(mapStateToProps, mapDispatchToProps)(BookList);
```

Comment:
- [ ] [OPTIONAL] If you want to learn the latest, and simplest, way of handling the connection of components with redux actions and states, check the documentation on [Redux hooks](https://react-redux.js.org/api/hooks) and [Redux ToolKit](https://redux-toolkit.js.org/).

Reply:
Thanks for the review and all suggestions uryelah.

Reply:
Thanks for the review and all suggestions uryelah.